index.js

```diff
@@ -31,13 +31,78 @@ SOFTWARE.
 var parseXML     = require('xml2js').parseString
   , url          = require('url')
   , EventEmitter = require('events').EventEmitter
+  , HttpProxyAgent = require('http-proxy-agent')
+  , HttpsProxyAgent = require('https-proxy-agent')
   ;
 
 // These numbers were obtained by measuring and averaging both using this module and the official speedtest.net
 var speedTestDownloadCorrectionFactor = 1.135
   , speedTestUploadCorrectionFactor   = 1.139
+  , proxyOptions = null
   ;
 
+// Search a proxy (http|https) in env with case-insensitive
+var proxyHttpEnv = findPropertiesInEnvInsensitive("HTTP_PROXY");
+var proxyHttpsEnv = findPropertiesInEnvInsensitive("HTTPS_PROXY");
+
+function findPropertiesInEnvInsensitive(prop) {
+  prop = (prop).toLowerCase();
+  for (var p in process.env) {
+    if (process.env.hasOwnProperty(p) && prop == (p).toLowerCase()) {
+      return process.env[p];
+    }
+  }
+  return null;
+}
+
+//set the proxy agent for each http request
+// priority : 
+// 1 - proxyOptions
+// 2 - proxyHttpEnv (HTTP_PROXY)
+// 3 - proxyHttpsEnv (HTTPS_PROXY)
+function proxy(options) {
+  var proxy = null;
+  var isSSL = false;
+  var haveHttp = false;
+  
+  if (!proxyHttpEnv && !proxyHttpsEnv && !proxyOptions) {
+    return;
+  }
+  // Test the proxy parameter first for priority
+  if (proxyOptions)
+  {
+    if (proxyOptions.startsWith("https:")) {
```

Comment:
Would you mind dropping all `startsWith` calls? They are ES6.

Reply:
by what do you want me to replace startwith? 

it is necessary to determine the protocol

Reply:
`if (proxyOptions.substr(0, 6) === "https:") {`

Reply:
i will change that thx

Reply:
done